framework/src/materials/MaterialPropertyStorage.C

```diff
@@ -385,8 +406,12 @@ void
 MaterialPropertyStorage::initProps(MaterialData & material_data,
                                    const Elem & elem,
                                    unsigned int side,
-                                   unsigned int n_qpoints)
+                                   unsigned int n_qpoints,
+                                   bool isNeighborOnInterface)
 {
+  // TODO _stateful_prop_id_to_prop_id contains the stateful properties belonging to the master
+  // blockRestricted a more smart way to solve the issue we have is to create a proper
+  // _stateful_prop_id_to_prop_id for the nieghbor block
```

Comment:
misspell

Reply:
Is there a large impediment to doing this? I don't want to introduce a lot of `TODO` statements in the code, but if there's a large overhead than this ok for now I would say

Reply:
I tried this route, but I'm I wasn't able to find a clean easy way to implement _stateful_prop_id_to_prop_id for the neighbour block. maybe you have some better idea.